test/catch2/wrappers/mock_connection.cpp

```diff
@@ -24,16 +24,24 @@ mock_connection::~mock_connection()
     if (_connection_impl->block_lock.initialized == 1)
         __wt_spin_destroy(nullptr, &_connection_impl->block_lock);
     __wt_free(nullptr, _connection_impl->chunkcache.free_bitmap);
+    /* setup_stats() used nullptr to allocate so use nullptr to free. */
+    __wt_stat_connection_discard(nullptr, _connection_impl);
     __wt_free(nullptr, _connection_impl);
 }
 
 std::shared_ptr<mock_connection>
-mock_connection::build_test_mock_connection()
+mock_connection::build_test_mock_connection(WT_SESSION_IMPL *session)
 {
     WT_CONNECTION_IMPL *connection_impl = nullptr;
+    mock_connection *mock_conn;
     utils::throw_if_non_zero(__wt_calloc(nullptr, 1, sizeof(WT_CONNECTION_IMPL), &connection_impl));
-    // Construct a Session object that will now own session.
-    return std::shared_ptr<mock_connection>(new mock_connection(connection_impl));
+
+    // Construct a mock_connection object that will now own connection_impl.
+    mock_conn = new mock_connection(connection_impl);
+
+    mock_conn->setup_stats(session);
```

Comment:
Should we just do this in the mock_connection constructor?

Reply:
OK.
I did it this way since it corresponds to mock_session::setup_block_manager_file_operations() and mock_session::setup_block_manager_session() which are outside the constructor and optional.
I don't think they should be optional. However, I did not change it.